coverage/src/Tooltip-Modal.js

```diff
@@ -0,0 +1,144 @@
+const modalTemplate = document.createElement('template');
+modalTemplate.innerHTML = `
+  <style>
+    .modal {
+      display: none;
+      position: fixed;
+      z-index: 1;
+      left: 0;
+      top: 0;
+      width: 100%;
+      height: 100%;
+      background-color: rgba(0, 0, 0, 0.4);
+    }
+
+    .modal-content {
+      background-color: #fefefe;
+      margin: 15% auto;
+      border: 1px solid #888;
+      width: 70%;
+      border-radius: 10px;
+    }
+
+    .modal-header,
+    .modal-footer {
+      display: flex;
+      align-items: center;
+      height: 60px;
+      padding: 0 20px;
+      box-sizing: border-box;
+    }
+
+    .modal-header {
+      justify-content: space-between;
+      border-bottom: 1px solid #ddd;
+    }
+
+    .modal-footer {
+      justify-content: flex-end;
+      border-top: 1px solid #ddd;
+    }
+
+    .modal-header h4 {
+      margin: 0;
+      font-size: 18px;
+    }
+
+    .modal-body {
+      text-align: justify;
+      padding: 0 30px;
+      overflow: auto;
+      font-size: 11pt;
+      max-height: 200px;
+      line-height: 1.5;
+    }
+
+    .close-icon {
+      color: #aaa;
+      float: right;
+      font-size: 20px;
+      font-weight: bold;
+    }
+
+    .close-icon:hover,
+    .close-icon:focus {
+      color: black;
+      text-decoration: none;
+      cursor: pointer;
+    }
+
+    button {
+      background-color: #2d8fc1;
+      color: white;
+      border: none;
+      padding: 5px 15px;
+      border-radius: 20px; 
+      cursor: pointer;
+    }
+    
+    button:hover {
+        background-color: #2d8fc1;
+        transform: scale(1.05);
+    }
+
+  </style>
+  <div class="modal">
```

Comment:
Would it be possible to use the relatively new HTML `dialog` element for this? https://developer.mozilla.org/en-US/docs/Web/HTML/Element/dialog

Might be too much trouble but worth looking into if you haven't already.

Reply:
Thanks for your reference. After reading it, the dialog has built-in method **showModal()** and **close()**, which is convenient to control the modal without creating the customized method. 

I noticed that you mentioned below to use CSS to control the modal. I tried to implement it this way. Please see #58

Reply:
@YangQi007 I'm not sure I understand. Did you try replacing the `<div>` with `<dialog>` and see if it works? That's what I meant.

Reply:
@anderspitman I thought you wanted me to use CSS to control the modal's show/hide. If we use <dialog> element, we will use the methods showModal() and close(). If we only use CSS, we can delete the customized methods in the modal component mentioned in your above comments.

I think using <dialog> will work. I can give it a try if we're not in a rush to deliver this tooltip feature.

Reply:
Generally it's good to use built-in HTML elements as long as they fit the purpose. I think in this case it's ok to use methods to control it, since there are other benefits to offset that tradeoff. I don't think it's worth spending a ton of time on, but if the built-in <dialog> element is relatively easy to use, it would be a good choice.

Reply:
Yes, the **dialog** element will be suitable for our case. I refactored the code, please check it out again.